### ISSUE TITLE
Alternative constructor that fails on non-existent parent directory

### ISSUE THREAD

**Issue by ryanavella** [derekdreery/lockfile-rs#8]
I would be interested in a constructor similar to `Lockfile::create` but which doesn't create parent directories. In my use-case, the user has control over the location where the lockfile is created, and it is an error if the user supplies a file path with a non-existent parent directory.

My currently workaround is to first verify that the parent directory exists before calling `Lockfile::create`, but this could result in a [race condition](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use).

I could get a PR ready quickly if needed, but I'm not sure what to name this constructor. Maybe `Lockfile::create_explicit` or `Lockfile::create_file` or `Lockfile::create_in_dir`? It is possible that my use-case has some overlap with issue #3, so another option would be to address that first.

**Comment by richard-uk1** [comment]
I think that this should be the default, so I've renamed `Lockfile::create` to `Lockfile::create_with_parents`, and added a new `Lockfile::create` that doesn't try to create parent dirs. Requires a major version bump (i.e. minor version bump since we are 0.x).

I've also updated to 2021 and made some other QOL improvements. Hope you're not stuck on an old compiler version!

Lemme know if there are any issues.
